### PR TITLE
Separate sync and async app webhooks

### PR DESCRIPTION
**Pull request by orzechdev**
I want to merge this change because... it adds sync and async app webhooks separation.

To review in: https://github.com/saleor/saleor-dashboard/pull/1635
⚠️ Waiting for backend PR to merge: https://github.com/saleor/saleor/pull/8523

**PR intended to be tested with API branch:**  `SALEOR-4665-separate-sync-async-webhooks`

### Screenshots

Zrzut ekranu 2021-11-30 o 14 09 43
Zrzut ekranu 2021-11-30 o 14 10 32

### Pull Request Checklist

1. [ ] This code contains UI changes
2. [ ] All visible strings are translated with proper context including data-formatting
3. [ ] Attributes `[data-test-id]` are added for new elements
4. [ ] Changes are mentioned in the changelog
5. [ ] The changes are tested in different browsers and in light/dark mode

### Test environment config

API_URI=https://saleor-4665-separate-sync-async-webhooks.api.saleor.rocks/graphql/

### CODE REVIEW

**Comment by patrys** [comment]
Task linked: [SALEOR-4637 Separate synchronous and asynchronous webhooks in UI](https://app.clickup.com/t/2549495/SALEOR-4637)

File: src/webhooks/components/WebhookCreatePage/WebhookCreatePage.tsx [hunk 1]
```diff
@@ -52,46 +63,71 @@ const WebhookCreatePage: React.FC<WebhookCreatePageProps> = ({
...
+        );
+
+        return (
+          <Container>
+            <Backlink onClick={onBack}>{appName}</Backlink>
+            <PageHeader
+              title={intl.formatMessage({
+                defaultMessage: "Create Webhook",
+                description: "header"
+              })}
```
Comment (Cloud11PL) [inline src/webhooks/components/WebhookCreatePage/WebhookCreatePage.tsx:95]: Move to a separate file

File: src/webhooks/components/WebhooksDetailsPage/WebhooksDetailsPage.tsx [hunk 2]
```diff
@@ -62,58 +69,83 @@ const WebhooksDetailsPage: React.FC<WebhooksDetailsPageProps> = ({
...
+        const handleSyncEventsSelect = createSyncEventsSelectHandler(
+          change,
+          data.syncEvents,
+          syncEventsChoices
+        );
+        const handleAsyncEventsSelect = createAsyncEventsSelectHandler(
+          change,
+          data.asyncEvents,
+          asyncEventsChoices
+        );
```
Comment (Cloud11PL) [inline src/webhooks/components/WebhooksDetailsPage/WebhooksDetailsPage.tsx:92]: Logic here seems to be the same as in `WebhookCreatePage.tsx`. I would extract it to a separate hook

Comment (orzechdev) [reply src/webhooks/components/WebhooksDetailsPage/WebhooksDetailsPage.tsx]: Indeed, ...I went further and removed the whole `WebhookCreatePage`, it is the same as `WebhookDetailsPage`

File: src/webhooks/components/WebhookDetailsPage/messages.ts [hunk 3]
```diff
@@ -0,0 +1,35 @@
+import { getStringOrPlaceholder } from "@saleor/misc";
+import { WebhookDetails_webhook } from "@saleor/webhooks/types/WebhookDetails";
+import { isUnnamed } from "@saleor/webhooks/utils";
+import { IntlShape } from "react-intl";
+import { defineMessages } from "react-intl";
+
+export const messages = defineMessages({
+  header: {
+    defaultMessage: "Unnamed Webhook Details",
```
[the remark this answers is not in the document]
Comment (orzechdev) [reply src/webhooks/components/WebhookDetailsPage/messages.ts:9]: Webhook does not require to be named.
Currently, there are 43 different enums, which might be used at once, do we want to concatenate them? If so, concatenating them could lead to a name with massive length. Do you mean it?

**Review by Cloud11PL — APPROVED** [review]

File: src/webhooks/components/WebhookDetailsPage/messages.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (jwm0) [reply src/webhooks/components/WebhookDetailsPage/messages.ts:9]: I've mistaken this for the enum event type, all good 👍

**Review by jwm0 — APPROVED** [review]

**Comment by maarcingebala** [comment]
@jwm0 @orzechdev This feature goes only to 3.1 so this PR is irrelevant.

**Pull request closed without merging**